xdsl/interactive/app.py

```diff
@@ -347,6 +389,49 @@ def get_query_string(self) -> str:
         )
         return f"xdsl-opt -p {query}"
 
+    def update_input_operation_count_tuple(self, input_text: str) -> None:
+        """
+        Function that updates the input_operation_datatable to display the operation
+        names and counts in the input text area.
+        """
+        if input_text == "":
+            self.input_operation_count_tuple = tuple()
+        else:
+            ctx = MLContext(True)
+            for dialect in get_all_dialects():
+                ctx.load_dialect(dialect)
+            module = Parser(ctx, input_text).parse_module()
+            self.input_operation_count_tuple = tuple(
+                count_number_of_operations(module).items()
+            )
+
+            self.input_operation_count_datatable.clear()
+            for k, v in self.input_operation_count_tuple:
+                self.input_operation_count_datatable.add_row(k, v)
```

Comment:
feels like this should be a "watch" method, similarly below

Reply:
but wait - do we want these variables to be reactive or not? I still don't understand. Right now they aren't reactive, but I suppose you want them to be? I explained in my comment above that I think making them reactive in this instance is a bit overkill, but okay I can do it.